### ISSUE TITLE
Nonadmins cannot operate on other users or roles

### ISSUE THREAD

**Issue by nicolaracco** [directus/directus#19128]
### Describe the Bug

I've created a "Staff" role which has admin_access=false, app_access=true and gave it permissions to:
- read all roles
- read all users whose role.admin_acess = false

However when this user access the backoffice they can see only their role. They can see all the users (even the ones in an admin role) in the list page, but can't open the details of no other user than theirs.

### To Reproduce

- Create a role with admin_access = false, app_access = true
- Give the role read permission on all roles
- Open the users page with a user of the newly created role and observe that only their role is present in the list

### Directus Version

v10.4.2

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by rijkvanzanten** [comment]
I can't reproduce this based on the steps. What database are you using? Would you be able to share a database dump with a minimal reproduction where this issue is happening? 

By default, a new role with app access already has read access to roles and their own user. Adding the rule to see all users where `role.admin_access = false` makes all users show up, the user detail pages render in readonly as per the update permissions.

**Comment by nicolaracco** [comment]
Thank you @rijkvanzanten for having tried to reproduce the issue. After your reply I tried to cleanup and repopulate both roles and permissions without success. I imported the production database locally without being able to reproduce the issue myself.
As a last attempt I connected to redis and gave a `FLUSHALL`. Problem solved.

I really have no idea what was causing this issue, I've spent some time trying to manipulate the data to cause it again without any success.
